File: algorithms/easy/separate_the_digits_in_an_array.cpp
class Solution {
public:
  vector<int> separateDigits(vector<int> &nums) {
    vector<int> ans;
    for (auto n : nums) {
      vector<int> arr;
      while (n > 0) {
        arr.push_back(n % 10);
        n /= 10;
      }
      ans.insert(ans.end(), arr.rbegin(), arr.rend());
    }
    return ans;
  }
};

File: algorithms/easy/separate_the_digits_in_an_array.rs
impl Solution {
    pub fn separate_digits(nums: Vec<i32>) -> Vec<i32> {
        let mut ans = Vec::new();
        for n in nums {
            let s = n.to_string();
            for b in s.as_bytes() {
                ans.push((b - b'0') as i32);
            }
        }
        ans
    }
}
